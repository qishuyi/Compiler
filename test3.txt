let max (x : int) (y : int) : int =
    if x > y then x else y ;;

let min (x : int) (y : int) : int =
    if x < y then x else y ;;

let main : int =
    let a : int = max 3 4 in
    let b : int = min 10 9 in
    let p : (int * int) = (a, b) in
    (fst p) + (snd p) ;;
