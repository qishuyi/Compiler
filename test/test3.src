3 +
(if
    3 <=0
 then
    true
 else
    false)