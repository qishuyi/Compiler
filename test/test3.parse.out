(+ 3 (if (<= 3 0) true false))